src/azul/plugins/__init__.py

```diff
@@ -546,6 +550,73 @@ def filter_stage(self) -> 'type[FilterStage]':
     def file_class(self) -> type['File']:
         raise NotImplementedError
 
+    def modify_hit(self,
+                   hit: MutableJSON,
+                   catalog: CatalogName,
+                   file_url_func: Callable,
+                   entity_type: str,
+                   source_ids: set[str],
+                   ) -> None:
+        self._inject_file_urls(hit, catalog, file_url_func, 'files')
+        entity = json_dict(one(json_list(hit[entity_type])))
+        source = json_dict(one(json_list(hit['sources'])))
+        source_id = source[self.special_fields.source_id]
+        entity[self.special_fields.accessible] = source_id in source_ids
+
+    @cache
+    def repository_plugin(self, catalog: CatalogName) -> 'RepositoryPlugin':
+        return RepositoryPlugin.load(catalog).create(catalog)
+
+    def _inject_file_urls(self,
+                          node: AnyMutableJSON,
+                          catalog: CatalogName,
+                          file_url_func: Callable,  # Can't use type FileUrlFunc due to circular import
+                          *path: str
```

Comment:
Nitpick: is there a reason for the `path` parameter to use `*`? Seems to me it would be easier to read the call sites if it were a regular list.